### PR TITLE
Refactor/340 refactor settings menu sliders to affect audio mixer groups instead of audio sources

### LINKED ISSUE #339: [BUG] Missing "select blip" sound effect inside ControlsMenu
## Describe the bug

Some button selections and interactions don't play the `SelectBlip2` sound effect played throughout the game when selecting UI elements.

## Steps To Reproduce

1. Navigate through the MainMenu and SettingsMenu - notice the sound effects
2. Navigate through the Controls Menu - notice the lack of sound effects

## Expected behavior

Sound effects play for all interactions, inside all menus.

## Screenshots

(turn on sound)

https://user-images.githubusercontent.com/1689033/218284660-9ad321a1-36e2-4077-8e2e-a9ecc469dd1d.mp4

## Additional context

Found in ea6a61fb92808bc40430066ef9244a3a958262be and 36877e0ca831deb47e122794827fc3d586b9f37e.

### LINKED ISSUE #340: Refactor SettingsMenu sliders to affect Audio Mixer Groups instead of Audio Sources
## Description

In the latest build (ea6a61fb92808bc40430066ef9244a3a958262be) Audio Sliders in the SettingsMenu affect Audio Source instances directly.

This has two downsides:
  - when adding more Audio Sources, they need a new slider or the player can't change the volume of that new Audio Source
  - can't share the SettingsMenu prefab across scenes: Audio Sources exist outside of the prefab; this requires resolving the mapping manually for each new scene

## Acceptance Criteria

This story will be finished
- when the Audio Sliders affect Audio Mixer Groups instead of Audio Source instances
  image

- all scenes use the same non-overridden SettingsMenu prefab:
  image

### LINKED ISSUE #341: [BUG] Audio slider ignores initial value of underlying AudioSource
## Describe the bug

When opening the SettingsMenu, the Audio Sliders use their default values (0.4), even though the underlying Audio Sources have a different volume.

#340 probably replaces the code that needs to change to fix this. Implementing #340 first won't implicitly fix this bug, but if the code ends up replaced anyway, probably best to refactor the implementation first and maybe resolve this in one go.

## Steps To Reproduce

1. Open the MainMenu Scene
2. Notice the value of `Audio/SfxAudioSource` property `Audio Source/Volume`
3. Start the Game and open the SettingsMenu
4. Notice the mis-matching value of the Slider
5. Change the slider
6. Notice that the value of the Slider and the underlying Audio Source now match

## Expected behavior

Audio Sliders have the same values as underlying Audio Source before making changes.

## Screenshots

image

## Additional context

Found in ea6a61fb92808bc40430066ef9244a3a958262be and 36877e0ca831deb47e122794827fc3d586b9f37e.

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

closes #340 and also closes #341 and closes #339 in the process.

Adds a new audio settings menu which instantiates sliders depending on the mixer groups in a given AudioMixer.

Also refactors the menu system somewhat so multiple input modules are not required for menus.

## Before/after screenshots and/or animated gif

Screenshot 2023-06-03 at 00 44 07
Sliders for each mixer group

## Testing instructions

Open the settings menu then open the audio menu. Do the sliders work? Do they play sound effects when selected? Do the sound effects play in the controls menu? Does it work in both the MainMenu and Game scenes?

## Additional information

- `[x]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource:

### CODE REVIEW

File: unity-ggjj/Assets/Scripts/Cheats/KonamiCodeHandler.cs [hunk 1]
```diff
@@ -22,8 +22,6 @@ private void Awake()
             cheatInput.Keyboard.ValidKeys.performed += ctx =>
             {
                 var pressedKey = NormalizeInput(ctx.control.name);
-
-                Debug.Log($"[KONAMICODE] Pressed key: {pressedKey}, expected: {_cheat[_currentCheatIndex]}");
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/Cheats/KonamiCodeHandler.cs:26]: Was considering whether or not to keep this, but setting a breakpoint to print output if there's a need to debug it instead is always an option, so I'm absolutely fine with removing it! :)

File: unity-ggjj/Assets/Scripts/UI/AudioSlider.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+using UnityEngine.UI;
+
+public class AudioSlider : MonoBehaviour
+{
+    [SerializeField] private AudioClip _onValueChangedSound;
+    [SerializeField] private float _updateSoundDelay;
+    
+    private const int MIN_DECIBELS = -80;
+    private const int MAX_DECIBELS = 20;
+    private const int DECIBEL_LOG_BASE = 10;
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/UI/AudioSlider.cs:13]: Was about to suggest moving it into the method where it's used, but stumbled across https://stackoverflow.com/a/49224799/1088653. Apparently const only offers a performance benefit on variable files, not inside methods. The more you know.

File: unity-ggjj/Assets/Scripts/UI/AudioSlider.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    {
+        return Mathf.Pow(10, volumeInDecibels / MAX_DECIBELS);
+    }
+
+    /// <summary>
+    /// Converts a value from 0 to 1 to a value from -80 to 0
+    /// I.E converts from a linear value to decibels used by AudioMixer
+    /// </summary>
+    /// <param name="linearVolume">The value to convert to decibels. Takes a value from 0 to 1</param>
+    private static float CalculateVolumeInDecibels(float linearVolume)
```
Comment (ViMaSter) [inline unity-ggjj/Assets/Scripts/UI/AudioSlider.cs:84]: Loooove the descriptive parameter names. 👌

**Review by ViMaSter — APPROVED** [review]
Suuuuper clean, as usual. 👍

Would be neat to have them covered by tests as well, but that probably only really makes sense, once the groups are cleaned up, now that music is actually no longer dynamic.

Is [MasterVolumeSlider.cs](https://app.codecov.io/gh/Studio-Lovelies/GG-JointJustice-Unity/pull/357/blob/unity-ggjj/Assets/Scripts/UI/MasterVolumeSlider.cs) safe to remove now? It appears to no longer be covered.

File: unity-ggjj/Assets/Scripts/Cheats/KonamiCodeHandler.cs (hunk 1 above)
Comment (PhoebeMitchell) [reply unity-ggjj/Assets/Scripts/Cheats/KonamiCodeHandler.cs:26]: It was mostly just annoying me as I was working on the main menu haha

**Pull request merged**